client/src/main/kotlin/io/spine/examples/shareaware/client/wallet/WalletPage.kt

```diff
@@ -94,32 +115,143 @@ private object WalletPageModel {
+     */
+    public fun replenishmentError(): StateFlow<Boolean> {
+        return replenishmentError
+    }
+
+    /**
+     * Cancels the state of the payment error occurred by the replenishment process.
+     */
+    public fun cancelReplenishmentError() {
+        replenishmentError.value = false
+    }
+
+    /**
+     * Sends the `ReplenishWallet` command to the server.
+     *
+     * @param ibanValue the IBAN of the user
+     * @param moneyAmount the amount of money to replenish the wallet
+     */
+    public fun replenishWallet(ibanValue: String, moneyAmount: String) {
+        val replenishWallet = ReplenishWallet
+            .newBuilder()
+            .buildWith(ibanValue, moneyAmount)
+        subscribeToWalletReplenished(replenishWallet.replenishment)
+        subscribeToReplenishmentError(replenishWallet.replenishment)
+        client.command(replenishWallet)
+    }
+
+    /**
+     * Subscribes to the `WalletReplenished` event.
+     *
+     * @param id the ID of the replenishment process
+     */
+    private fun subscribeToWalletReplenished(id: ReplenishmentId) {
+        val replenishmentIdField = WalletReplenished.Field.replenishment()
+        client.subscribeOnce(
+            WalletReplenished::class.java,
+            eq(replenishmentIdField, id)
+        ) {
+            replenishmentError.value = false
+        }
+    }
+
+    /**
+     * Subscribes to the `MoneyCannotBeTransferredFromUser` event that signals about
+     * failure in the payment system, during the wallet replenishment process.
+     *
+     * @param id the ID of the replenishment process
+     */
+    private fun subscribeToReplenishmentError(id: ReplenishmentId) {
+        val replenishmentIdField = MoneyCannotBeTransferredFromUser.Field.replenishment()
+        client.subscribeOnce(
+            MoneyCannotBeTransferredFromUser::class.java,
+            eq(replenishmentIdField, id)
+        ) {
+            replenishmentError.value = true
+        }
+    }
+
+    /**
+     * Returns the command to replenish the wallet.
+     *
+     * @param ibanValue the IBAN of the user
+     * @param amount the amount of money to replenish the wallet
+     */
+    private fun ReplenishWallet.Builder.buildWith(ibanValue: String, amount: String): ReplenishWallet {
+        return this
+            .setReplenishment(ReplenishmentId.generate())
+            .setWallet(client.wallet())
+            .setIban(ibanValue.asIban())
+            .setMoneyAmount(amount.asMoney())
+            .vBuild()
+    }
+
+    /**
+     * Returns a new `Money` object using this `String` to construct it.
+     *
+     * This `String` must be written as a number with decimal point.
+     */
+    private fun String.asMoney(): Money {
+        val parts = this.split('.')
+        val units = parts[0].toLong()
+        val nanos = if (parts.size == 2) parts[1].toInt() else 0
+        return Money
+            .newBuilder()
+            .setCurrency(Currency.USD)
```

Comment:
Let's mention this in the documentation.

Possibly, also rename the method `asUsd()`.